### PR TITLE
Gateway integration tests: final integrity check

### PR DESCRIPTION
**Pull request by janpriessnitz**

### CODE REVIEW

**Comment by janpriessnitz** [comment]
@cernvm-bot cloudtest

**Comment by janpriessnitz** [comment]
@cernvm-bot cloudtest nodestroy

File: test/test_functions [hunk 1]
```diff
@@ -1140,6 +1140,18 @@ check_repository() {
   return 0
 }
 
+check_repo_integrity() {
+  local repo=$1
+  for f in `find /cvmfs/${repo} -type f`; do
```
Comment (jblomer) [inline test/test_functions:1145]: More compact:

    if find /cvmfs/${repo} -type f -exec cat {} >/dev/null \; ; then

**Comment by janpriessnitz** [comment]
@cernvm-bot cloudtest

**Comment by janpriessnitz** [comment]
@cernvm-bot cloudtest

File: test/test_functions (hunk 1 above)
Comment (janpriessnitz) [reply test/test_functions:1145]: I wanted to print out the corrupted file, which would be messy with the compact version. Is this really necessary?

**Pull request merged**